### ISSUE TITLE
Alternative for retrieving the images if allow_url_fopen = off / file_get_contents not allowed

### ISSUE THREAD

**Issue by thorstenBorn-cosmoblonde** [pixx-io/typo3#9]
Plugin: pixxio/pixxio-extension
Version: 2.0.0

Symptom: Error message when selecting an image from the pixxio iFrame that the file could not be saved to the target directory (target directory = empty string).

Cause: The error message is misleading because there is no possibility of outputting a path. A variable with the target path is simply missing in the code. 
In our case, the reason for the error message was that file_get_contents is simply not allowed for URLs. Not knowing yet whether enabling it in this case will be allowed at our customer (self-managed server/self-hosted), here is a possibility to call the file via curl if allow_url_fopen is disabled. 

Solution: Use curl as fallback - this solution is only briefly tested. If necessary, error handling must be added.
Additional tasks: put curl to requirements in composer.json

Classes/Controller/FilesController.php # 597
```php
$uploaded = file_put_contents($absFileIdentifier, file_get_contents($url));
```
replace by

```php
if( ini_get('allow_url_fopen') ) {
    $uploaded = file_put_contents($absFileIdentifier, file_get_contents($url));
} else {
    $ch = curl_init($url);
    $fp = fopen($absFileIdentifier, 'wb');
    curl_setopt($ch, CURLOPT_FILE, $fp);
    curl_setopt($ch, CURLOPT_HEADER, 0);

    $uploaded = true;
    if( ! curl_exec($ch)) {
        // error throw here would be fine
        $uploaded = false;
    }
    curl_close($ch);
    fclose($fp);
}
```

**Comment by machwert** [comment]
Thanks @thorstenBorn-cosmoblonde!
Commited to dev branch and will be merged to master after testing.
I added ext-curl as requirement in composer.json to be sure curl is running.

**Comment by thorstenBorn-cosmoblonde** [comment]
Thank you very much. 
It just came to me that the curl request is not needed if allow_url_fopen is switched on. And so I don't know if it's good as a "hard" request... But I don't know of any way to formulate a "conditional request", especially since allow_url_fopen is, I think, a Composer request. But because we deploy automatically, the setting is correct on the gitlab server, but it's turned off on the target server. 
Maybe you need to observe this to see if there is any feedback from other users.

**Comment by rogervanwile** [comment]
Hi @thorstenBorn-cosmoblonde ,

thanks for creating the issue. We published a new version 2.0.1 with changes related to the issue. Please update your installation.
